### ISSUE TITLE
Backbone models shoudn't have getters and setters for the special `id` attribute ?

### ISSUE THREAD

**Issue by loicraux** [Microsoft/tslint-microsoft-contrib#84]
This is for discussion...

`model.id` is a special property of models, manipulated (set and get) at several places in backbone.js lib.
So, not to have issues at runtime, :
- either a Backbone Model must have both Typescript getter and Typescript setter for this id property
- or must not define getter and setter for id and rather make use of `Backbone.Model.prototype.idAttribute` and/or the new `Backbone.Collection.prototype.modelId`...

What is your opinion on this one ? Imho, I prefer the second option.
Should we create a rule for it ?

**Comment by HamletDRC** [comment]
id is public on Backbone.Model. Why would anyone add getters and setters? 

Can you show some examples of code that violates this and code that is OK?

**Comment by loicraux** [comment]
People would add getter to map from another key/attr to id :

```typescript
class MyModel extends Backbone.Model {
  public get id(): string {
    return this.get('someOtherAttr');
  }
}
```

But : 

 - This code above is likely to break at runtime with the latest backbone version (new places where we have `this.id = ...` statement internally in backbone.
 - We already have `Backbone.Model.prototype.idAttribute` which purpose is exactly to map another key/attr to id. (Plus also now `Backbone.Collection.prototype.modelId` func that can be overridden if you deal with heterogeneous Collections)

**Comment by HamletDRC** [comment]
What should we do here? Implement it? 

It would be hard to know that a class extends Backbone.Model because we use subclasses of subclasses. Could be done though.

**Comment by HamletDRC** [comment]
@loicraux do you still want this rule or do you think it is too general given that we cannot know if a class is a subclass of Backbone.Model or not?

**Comment by JoshuaKGoldberg** [comment]
Hey @loicraux I'm going to go ahead and close this issue for housekeeping. Post here if you still want this!
